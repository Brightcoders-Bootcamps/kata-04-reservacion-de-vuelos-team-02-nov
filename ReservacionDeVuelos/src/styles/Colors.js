export const colors = {
    Purple: '#6472ee',
    ligthGray: '#b6b7ba',
    white: '#FFF',
    grayTitle: '8c8c8c'
}