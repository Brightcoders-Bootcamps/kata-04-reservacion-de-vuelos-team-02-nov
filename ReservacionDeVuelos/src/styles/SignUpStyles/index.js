import React from 'react';
import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  title: {
    fontSize: 20,
    color: '#2D02FF',
    fontWeight: 'bold',
    marginBottom: 20,
  },
  button: {
    height: 40,
  },
  SignUpContainer: {
    margin: 10,
    padding: 10,
  },
  textInput: {
    height: 40,
    borderColor: 'gray',
    borderWidth: 1,
    marginBottom: 10,
  },
  formText: {
    marginBottom: 10,
    fontSize: 16,
  },
  checkBoxContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
    marginTop: 5,
  },
  btnSignupContainer: {
    marginTop: 15,
  },
  btnSignup: {
    backgroundColor: '#b6b7ba',
    padding: 12,
    borderRadius: 10,
    marginBottom: 5,
    flexDirection: 'row',
    justifyContent: 'center',
  },
  btnSignupText: {
    textAlign: 'center',
    color: '#FFF',
    fontWeight: 'bold',
    fontSize: 18,
  },
  or: {
    textAlign: 'center',
    marginBottom: 10,
    marginTop: 10,
    color: '#b6b7ba',
  },
  LogInText: {
    color: 'blue'
  },
  loginTextContainer: {
    textAlign: 'center',
    marginTop: 20, 
  },
  googleIcon: {
    width: 20,
    height: 20,
  },
  characters: {
    marginBottom: 20,
  },
  checkboxFather: {
    marginBottom: 15,
  },
});
