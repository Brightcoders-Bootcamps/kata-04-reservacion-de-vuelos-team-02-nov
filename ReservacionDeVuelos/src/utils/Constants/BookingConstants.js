const TitleOrigin = 'Where are you now?';
const TitleDestination = 'Where will you be flying to?';
const TextInput = 'Select location';
const BtnText = 'Next'; 
const SelectDate = 'Select Date';
const Passengers = 'How many passengers?';

export const BookingConstants = {
    TitleOrigin,
    TitleDestination,
    TextInput,
    BtnText, 
    SelectDate,
    Passengers,
};