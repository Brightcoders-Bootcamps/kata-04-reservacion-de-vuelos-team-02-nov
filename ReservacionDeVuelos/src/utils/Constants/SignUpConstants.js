const SignUpTitle = 'Sign Up';
const NameInputTitle = 'First Name';
const EmailInputTitle = 'Email *';
const PasswordInputTitle = 'Password *';
const TermsCheckboxTitle = 'I agree to the Terms and Privacy Policy.';
const UpdatesCheckboxTitle = 'Subscribe for select product updates.';
const CharactersMessage = 'Use 8 or more characters with a mix of letters, numbers, and simbols.';
const SignUpButton = 'Sign Up';
const GoogleSignUpButton = 'Sign Up with Google';
const HasAnAccount = 'Already have an account?';
const LogIn = 'Log In';
export const SignUpConstants = {
  SignUpTitle,
  NameInputTitle,
  EmailInputTitle,
  PasswordInputTitle,
  TermsCheckboxTitle,
  UpdatesCheckboxTitle,
  CharactersMessage,
  SignUpButton,
  GoogleSignUpButton,
  HasAnAccount,
  LogIn,
};
