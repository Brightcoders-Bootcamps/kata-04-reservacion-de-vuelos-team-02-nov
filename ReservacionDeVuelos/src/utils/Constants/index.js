export * from './SignUpConstants';
export * from './FlightsListConstants'
