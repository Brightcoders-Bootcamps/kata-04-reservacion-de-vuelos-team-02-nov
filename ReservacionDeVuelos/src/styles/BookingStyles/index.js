import {StyleSheet} from 'react-native';
import {colors} from '../Colors';

export const styles = StyleSheet.create({  
    FlightTitle: {
        color: colors.black,  
        fontSize: 35,
        fontWeight: 'bold',
        width: 300,
        marginBottom: 80,
    },
    BookingTextInput: { 
       height: 70,
       borderBottomColor: colors.black,
       borderBottomWidth: 1,
       fontSize: 20,
       borderBottomColor: colors.ligthGray,
    },
    BookingButton: {
        backgroundColor: colors.ligthGray,
        padding: 12,
        borderRadius: 10,
        justifyContent: 'center',
    },
    BookingButtonText: {
        textAlign: 'center',
        color: colors.white,
        fontWeight: 'bold',
        fontSize: 18,
    },
    ScreenContainer: {   
        marginTop: 50,
        marginBottom: 50,
        marginLeft: 20,
        marginRight: 20,
    },
    ContainerInformation: {
        flexDirection: 'column',
        justifyContent: 'space-between',
        height: '50%',
    },  
    ContainerFlightInformation: {
        flexDirection: 'column',
        justifyContent: 'space-between',
        height: '50%', 
    },  
    ContainerForm: {
        flexDirection: 'column',
        justifyContent: 'space-between',
        height: '50%',
    },
    BookingInformation: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        borderBottomWidth: .5,
        borderColor: colors.ligthGray,
        height: 70, 
    },
    InformationFlightsTitle: {
        fontSize: 25,
        fontWeight: 'bold',
    },
    InformationFlightsSubtitle: {
        color: colors.ligthGray,
        marginTop: 5,
    },
    left:{
        alignSelf: 'flex-start',
    },
    plane: {
        alignSelf: 'center',
    },
});
