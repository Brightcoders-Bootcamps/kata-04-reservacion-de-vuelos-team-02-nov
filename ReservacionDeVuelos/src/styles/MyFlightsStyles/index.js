import {StyleSheet} from 'react-native';
import {colors} from '../Colors'

export const styles = StyleSheet.create({
  ListContainer: {
    padding: 15,
  },
  TopContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  middleContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderBottomWidth: 2,
    borderBottomColor: colors.grayTitler
  },
  bottomContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },

  country: {
    fontSize: 30,
    fontWeight: 'bold',
  },
  city: {
    color: colors.greyTitle,
  },
  border: {
    borderBottomColor: 'black',
    borderBottomWidth: 1,
  }
});
