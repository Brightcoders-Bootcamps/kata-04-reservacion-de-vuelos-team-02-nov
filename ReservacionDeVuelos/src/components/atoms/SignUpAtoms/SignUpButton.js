import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import {colors} from '../../../styles/Colors';
import {styles} from '../../../styles/SignUpStyles';
import {SignUpConstants} from '../../../utils/Constants';

const SignUpButton = ({isButtonEnabled, formValues, signUpSubmit}) => {
  return (
    <TouchableOpacity
      testID="SignUpBtn"
      style={[
        styles.btnSignup,
        {backgroundColor: isButtonEnabled ? colors.Purple : colors.ligthGray},
      ]}
      onPress={() => {
        isButtonEnabled
          ? signUpSubmit(formValues)
          : alert('Invalid Credentials');
      }}>
      <Text testID="btnTitle" style={styles.btnSignupText}>
        {SignUpConstants.SignUpButton}
      </Text>
    </TouchableOpacity>
  );
};

export default SignUpButton;
