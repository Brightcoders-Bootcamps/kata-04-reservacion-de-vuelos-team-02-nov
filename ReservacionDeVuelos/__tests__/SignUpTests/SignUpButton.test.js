import React from 'react';
import SignUpButton from '../../src/components/atoms/SignUpAtoms/SignUpButton';
import {shallow} from 'enzyme';
import {SignUpConstants} from '../../src/utils/Constants';
import {colors} from '../../src/styles/Colors';
let wrapper;
beforeEach(() => {
  wrapper = shallow(<SignUpButton />);
});

describe('SignUpButton component renders as expected', () => {
  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('button exists', () => {
    const btn = wrapper.findWhere(
      (node) => node.prop('testID') === 'SignUpBtn',
    );
    expect(btn.length).toBe(1);
  });
  test('button contains a title', () => {
    const btnTitle = wrapper.findWhere(
      (node) => node.prop('testID') === 'btnTitle',
    );
    expect(btnTitle.length).toBe(1);
  });
  test('button title matches with our constant', () => {
    const btnTitle = wrapper.findWhere(
      (node) => node.prop('testID') === 'btnTitle',
    );
    expect(btnTitle.props().children).toBe(SignUpConstants.SignUpButton);
  });
});

describe('Funcionality when is not enabled', () => {
  const mockFormValues = {
    name: '',
    email: '',
    password: '',
    termsCheckbox: false,
    updatesCheckbox: false,
    validName: false,
    validEmail: false,
    validPassword: false,
    validatedName: false,
    validatedEmail: false,
    validatedPassword: false,
  };
  const signUpSubmitMock = jest.fn();
  const wrapper = shallow(
    <SignUpButton
      isButtonEnabled={false}
      formValues={mockFormValues}
      signUpSubmit={signUpSubmitMock}
    />,
  );
  test('buttton color matches with our color constant', () => {
    const btn = wrapper.findWhere(
      (node) => node.prop('testID') === 'SignUpBtn',
    );
    expect(btn.props().style[1].backgroundColor).toEqual(colors.ligthGray);
  });
  test('submit function is not called', ()=>{
    const btn = wrapper.findWhere(
      (node) => node.prop('testID') === 'SignUpBtn',
    );
    btn.props().onPress();
    wrapper.update();
    expect(signUpSubmitMock).not.toHaveBeenCalled();
  })
});

describe('Funcionality when is enabled', () => {
  const mockFormValues = {
    name: 'juan',
    email: 'juan@mail.com',
    password: 'mm23mm222',
    termsCheckbox: true,
    updatesCheckbox: true,
    validName: true,
    validEmail: true,
    validPassword: true,
    validatedName: true,
    validatedEmail: true,
    validatedPassword: true,
  };
  const signUpSubmitMock = jest.fn();
  const wrapper = shallow(
    <SignUpButton
      isButtonEnabled={true}
      formValues={mockFormValues}
      signUpSubmit={signUpSubmitMock}
    />,
  );
  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('buttton color matches with our color constant', () => {
    const btn = wrapper.findWhere(
      (node) => node.prop('testID') === 'SignUpBtn',
    );
    expect(btn.props().style[1].backgroundColor).toEqual(colors.Purple);
  });
  test('onPress event gtsc called', () => {
    const btn = wrapper.findWhere(
      (node) => node.prop('testID') === 'SignUpBtn',
    );
    btn.props().onPress();
    wrapper.update();
    expect(signUpSubmitMock).toHaveBeenCalled();
  });
});
