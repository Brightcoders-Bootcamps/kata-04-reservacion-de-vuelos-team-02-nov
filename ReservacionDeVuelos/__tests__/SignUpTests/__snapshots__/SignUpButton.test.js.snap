// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Funcionality when is enabled snapshot 1`] = `ShallowWrapper {}`;

exports[`SignUpButton component renders as expected renders correctly 1`] = `ShallowWrapper {}`;
